### ISSUE TITLE
I am trying to obtain the same result with pytorch.

### ISSUE THREAD

**Issue by nile649** [Puzer/stylegan#10]
I did the encoding like the author of this github has mentioned.
VGG = Perceptual model
G = Generator model
G.style() is the fully connected encoder in generator which gives dlatent code.

First question
After optimizing perceptual loss between G and VGG, do I need to save the weights of G.style(), or pass random vector to G.style() save the output from G.style() which is dlatent.

Second Question
If the first approach is right, than I proceed by getting attributes of the face using microsoft cognitive api. I tried on just 860 images to begin with.
I do get the result but it is wrong.
![image](https://user-images.githubusercontent.com/32990329/55182756-c24e4680-5164-11e9-967a-b47d8bc9df54.png)

**Comment by nile649** [comment]
How is z latent learned with perceptual loss? when generator has no gradient to backpropagate.

**Comment by nile649** [comment]
![image](https://user-images.githubusercontent.com/32990329/55182849-f1fd4e80-5164-11e9-81a4-26e8028a975e.png)
![image](https://user-images.githubusercontent.com/32990329/55182859-f7f32f80-5164-11e9-8e75-4ba0829804ec.png)

**Comment by nile649** [comment]
Smile direction.
![image](https://user-images.githubusercontent.com/32990329/55183805-00e50080-5167-11e9-9663-65a1bbd0391e.png)

![image](https://user-images.githubusercontent.com/32990329/55183823-0a6e6880-5167-11e9-9f5a-367e22d104ff.png)

**Comment by bayminbx** [comment]
@nile649  Do  you obtain the same result with pytorch? Can you share you code???Thank you.

**Comment by drprajapati** [comment]
https://github.com/jacobhallberg/pytorch_stylegan_encoder/blob/master/train_image_to_latent_model.ipynb
